src/main/java/bitflow4j/steps/database/WriteHistogramToElasticsearch.java

```diff
@@ -57,33 +85,39 @@ public WriteHistogramToElasticsearch(String host, int port1, int port2, String i
             }
             data.endObject();
 
-            //Index stands for Putting data
             request.source(data);
 
             IndexResponse response = client.index(request, RequestOptions.DEFAULT);
-            printResponse(response);
+            printResponse(response, sample);
         }
-
         return batch;
     }
-    private void printResponse(IndexResponse indexResponse){
+
+    private void printResponse(IndexResponse indexResponse, Sample sample) {
         String index = indexResponse.getIndex();
         String id = indexResponse.getId();
         if (indexResponse.getResult() == DocWriteResponse.Result.CREATED) {
             // Object was created in Database
+            logger.log(Level.FINE, "Document was created.");
         } else if (indexResponse.getResult() == DocWriteResponse.Result.UPDATED) {
-            logger.log(Level.INFO, "Rewritten existing document");
+            logger.log(Level.FINE, String.format("Rewritten existing document: %s", toString(sample)));
         }
         ReplicationResponse.ShardInfo shardInfo = indexResponse.getShardInfo();
         if (shardInfo.getTotal() != shardInfo.getSuccessful()) {
-            // Something with shards.
+            logger.log(Level.WARNING, String.format("Total-Shards (%s) did not match successful Shards (%s): %s",
+                    shardInfo.getTotal(), shardInfo.getSuccessful(), toString(sample)));
         }
         if (shardInfo.getFailed() > 0) {
             for (ReplicationResponse.ShardInfo.Failure failure :
                     shardInfo.getFailures()) {
                 String reason = failure.reason();
-                logger.log(Level.SEVERE, String.format("Failure: %s ", failure.reason()));
+                logger.log(Level.SEVERE, String.format("Failure (%s):\n %s ", toString(sample), failure.reason()));
             }
         }
     }
+
+    public String toString(Sample sample) {
```

Comment:
Sorry for being confusing about this: we need a toString() method without parameters, which overrides the default toString() methods present in all Java Objects. You can simply remove the sample parameter and print the unresolved identifierTemplate.